lib/beacon/component_definitions.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Beacon.ComponentDefinitions do
+  @moduledoc """
+  The ComponentDefinitions context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Beacon.Content.ComponentDefinition
+
+  alias Beacon.Repo
+
+  @doc """
+  Returns the list of component definitions.
+
+  ## Examples
+
+      iex> list_component_definitions()
+      [%ComponentDefinitions{}, ...]
+
+  """
+  def list_component_definitions do
+    ComponentDefinition |> Repo.all()
+  end
+
+  @doc """
+  Gets a single ComponentDefinition.
+
+  Raises `Ecto.NoResultsError` if the ComponentDefinition does not exist.
+
+  ## Examples
+
+      iex> get_component_definition!(123)
+      %ComponentDefinition{}
+
+      iex> get_component_definition!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_component_definition!(id, preloads \\ []), do: ComponentDefinition |> Repo.get!(id) |> Repo.preload(preloads)
```

Comment:
```suggestion
  def get_component_definition!(id, preloads \\ []) do
    ComponentDefinition
    |> Repo.get!(id)
    |> Repo.preload(preloads)
  end
```
`do/end` block is preferred for pipelines

Reply:
Gotcha.